src/views/become-a-scratcher/become-a-scratcher.jsx

```diff
@@ -0,0 +1,585 @@
+            </Modal>
+            <span>
+                <span className="section">{sectionText}</span>
+            </span>
+            <Button
+                onClick={() => setShowModal(true)}
+                className={`finish-later ${secondary ? 'secondary-finish-later' : ''}`}
+            >
+                <FormattedMessage
+                    id={'becomeAScratcher.buttons.finishLater'}
+                />
+            </Button>
+        </div>
+    );
+};
+OnboardingHeader.propTypes = {
+    user: PropTypes.shape({
+        id: PropTypes.number,
+        thumbnailUrl: PropTypes.string,
+        username: PropTypes.string
+    }),
+    sectionText: PropTypes.string,
+    secondary: PropTypes.bool
+};
+
+const OnboardingNavigation = ({currentPage, totalDots, onNextPage, onBackPage, nextButtonText}) => {
+    const dots = [];
+
+    if (currentPage && totalDots){
+        for (let i = 0; i < totalDots - 1; i++){
+            // First two pages don't have dots
+            dots.push(<div className={`dot ${currentPage === i + 2 && 'active'}`} />);
+        }
+    }
+    
+    return (
+        
+        <div className="navigation">
+            <Button onClick={onBackPage}>
+                <img
+                    className="left-arrow"
+                    alt=""
+                    src="/images/onboarding/left-arrow.svg"
+                />
+                <span className="navText">
+                    <FormattedMessage
+                        id={'becomeAScratcher.buttons.back'}
+                    />
+                </span>
+            </Button>
+            {(currentPage && totalDots) &&
+            <div className="dotRow">
+                {dots}
+            </div>}
+            <Button onClick={onNextPage}>
+                <span className="navText">
+                    {nextButtonText || <FormattedMessage id={'becomeAScratcher.buttons.next'} />}
+                </span>
+                <img
+                    className="right-arrow"
+                    alt=""
+                    src="/images/onboarding/right-arrow.svg"
+                />
+            </Button>
+        </div>
+
+    );
+};
+OnboardingNavigation.propTypes = {
+    currentPage: PropTypes.number,
+    totalDots: PropTypes.number,
+    onNextPage: PropTypes.funct,
+    onBackPage: PropTypes.funct,
+    nextButtonText: PropTypes.string
+};
+
+const BecomeAScratcher = ({user, invitedScratcher, scratcher, sessionStatus}) => {
+    const [currentPage, setCurrentPage] = useState(0);
+    const [hoorayAppear, setHoorayAppear] = useState(false);
+    const [showConfetti, setShowConfetti] = useState(true);
+    const [showPromotionError, setShowPromotionError] = useState(false);
+    
+    const {width, height} = useWindowSize();
+
+    const lastPage = 11;
+
+    const handlePromoteToScratcher = onSuccess => {
+        api({
+            host: '',
+            uri: `/users/${user.username}/promote-to-scratcher/`,
+            method: 'GET'
```

Comment:
Shouldn't this be a POST endpoint?

Reply:
> Shouldn't this be a POST endpoint?

Probably, so people don't open the link accidentally. What is it currently?

Reply:
> > Shouldn't this be a POST endpoint?
> 
> Probably, so people don't open the link accidentally. What is it currently?

It’s a GET endpoint.

Reply:
> > > Shouldn't this be a POST endpoint?
> > 
> > 
> > Probably, so people don't open the link accidentally. What is it currently?
> 
> It’s a GET endpoint.

Huh.

Reply:
we believe the endpoint's secure, thanks for the check!